pydicom/dataset.py

```diff
@@ -1226,17 +1278,15 @@ def pop(self, key, *args):
         """
         try:
             tag = Tag(key)
-        except (ValueError, OverflowError):
+        except Exception:
             return self._dict.pop(key, *args)
```

Comment:
Won't this always raise `KeyError`? If *key* isn't convertible into a `BaseTag` then it can't be in `_dict`